buku.py

```diff
@@ -1740,11 +1738,165 @@ def exportdb(self, filepath, taglist=None):
+            )
+            title_data = res.fetchone()
+            if title_data:
+                title = title_data[0]
+            else:
+                title = ''
+            tags = self.parse_tags(bookmark_tags)
+            self.add_rec(url, title, tags)
+        try:
+            cur.close()
+            conn.close()
+        except Exception:
+            print('Error here')
+
+    def get_firefox_profile_name(path):
+        '''List folder and detect default firefox profile name.
+        :return: profile name
+        '''
+        names = os.listdir(path)
+        profile = [name[:-8] for name in names if name.endswith('.default')][0]
+        return profile
+
+    def import_from_browser(self):
+        '''Import bookmarks from a browser default database file.
+        Supports Firefox and Google Chrome.
+
+        :return: True on success, False on failure
+        '''
+        if sys.platform.startswith('linux'):
+            GC_BOOKMARK_DATABASE_PATH = '~/.config/google-chrome/Default/Bookmarks'
+
+            DEFAULT_FF_FOLDER = os.path.expanduser('~/.mozilla/firefox')
+            profile = self.get_firefox_profile_name(DEFAULT_FF_FOLDER)
+            FF_BOOKMARK_DATABASE_PATH = (
+                '~/.mozilla/firefox/{}.default/places.sqlite'.format(profile)
+            )
+        elif sys.platform == 'darwin':
+            GC_BOOKMARK_DATABASE_PATH = (
+                '~/Library/Application Support/Google/Chrome/Default/Bookmarks'
+            )
+            DEFAULT_FF_FOLDER = os.path.expanduser('~/Library/Application Support/Firefox')
+            profile = self.get_firefox_profile_name(DEFAULT_FF_FOLDER)
+            FF_BOOKMARK_DATABASE_PATH = (
+                '~/Library/Application Support/Firefox/{}.default/places.sqlite'.format(profile)
+            )
+
+        elif sys.platform == 'win32':
+            username = os.getlogin()
+            GC_BOOKMARK_DATABASE_PATH = (
+                'C:/Users/{}/AppData/Local/Google/Chrome/User Data/Default/Bookmarks'.format(username)
+            )
+            DEFAULT_FF_FOLDER = 'C:/Users/{}/AppData/Roaming/Mozilla/Firefox/Profiles'.format(username)
+            profile = self.get_firefox_profile_name(DEFAULT_FF_FOLDER)
+            FF_BOOKMARK_DATABASE_PATH = (
+                os.path.join(DEFAULT_FF_FOLDER, '{}.default/places.sqlite'.format(profile))
+            )
+        try:
+            webbrowser.get('google-chrome')
+            bookmarks_database = os.path.expanduser(GC_BOOKMARK_DATABASE_PATH)
+            self.load_chrome_database(bookmarks_database)
+
+        except Exception as e:
+            logerr(e)
+            logerr('Could not detect `google-chrome\' browser')
+
+        try:
+            webbrowser.get('firefox')
+            bookmarks_database = os.path.expanduser(FF_BOOKMARK_DATABASE_PATH)
+            self.load_firefox_database(bookmarks_database)
+
+        except Exception as e:
+            logerr(e)
```

Comment:
iirc in this case no because the program is not exit yet.

there is other catch exception with error message and maybe better to make it similar

https://github.com/jarun/Buku/blob/master/buku.py#L522

```python
        except Exception as e:
            logerr('initdb(): %s', e)
            sys.exit(1)
```
just remove sys.exit, change the message and keep single `loggerr` call